src/main/java/task/TaskList.java

```diff
@@ -0,0 +1,124 @@
+package task;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import duke.Storage;
+import duke.Ui;
+
+/**
+ * Implementation of task list, which is a list of tasks
+ */
+public class TaskList {
+
+    private final ArrayList<Task> tasks;
+
+    /**
+     * Constructor, creates new task list
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Constructor, takes in and uses existing task lists
+     *
+     * @param tasks Existing task list to use
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Prints the number of tasks in the list
+     */
+    public void printSize() {
+        System.out.println(Ui.OUTPUT_DISPLAY
+                + (tasks.size() == 1
+                    ? "There is 1 task in your list"
+                    : "There are " + tasks.size() + " tasks in your list"));
+    }
+
+    /**
+     * Adds a task into the task list
+     *
+     * @param task Task to add
+     */
+    public void add(Task task) {
+        Optional.ofNullable(task)
```

Comment:
Really obscure way of doing a null check haha

Reply:
:P